### DISCUSSION TITLE
How to avoid "Probably Zeno behaviour" error, when simulating untimed model for too long?

### DISCUSSION THREAD

**Discussion by AstridHornBrorholt** [UPPAALModelChecker/UPPAAL-Meta#210] [Q&A]
I have a model that does not make use of clocks, which I want to run for a very long time. 
When I exceed 100 000 discrete jumps, it errors out with the following message: 

    Probably Zeno behavior (infinitely many discrete transitions in finite amount of time):
    Executed more than 100000 discrete transition while delaying only 0 time units.
    State: ( Process._id0 ) acc=71#t(0)=0#time=0

The error message is correct in the sense that I perform a lot of discrete jumps without advancing any clocks, but in this instance I want to do so. 

To quickly reproduce the issue, I have created this toy example: 
[coinflipper.zip](https://github.com/UPPAALModelChecker/UPPAAL-Meta/files/11869261/coinflipper.zip)

![image](https://github.com/UPPAALModelChecker/UPPAAL-Meta/assets/11016262/8d83784a-1ee6-4dcc-8a8f-80e4c3fda47d)

This query is fine:

    /* OK */ E[#<=10000;1] (max: acc)

But this query causes an error:

    /* Bad */ E[#<=100000;1] (max: acc)

Is there some sort of trick to avoid this error?

**Comment by thorulf4** [comment]
I don't know of any way to disable zeno checking. However if you do a delay beyond some non zero constant (default is 0.01) then you won't get zeno behaviour errors. I would recommend adding a clock x and forcing a delay of 1 during each toss of the coin. 
![image](https://github.com/UPPAALModelChecker/UPPAAL-Meta/assets/10059450/daa0b691-9cb8-41c0-b37f-d111a3dcf94d)

With this modification I successfully ran with 100.000.000 steps

If these modifications are not applicable to your use case please let us know we might be able to figure something out.

**Reply by thorulf4** [reply]
You may also use rate of exponential instead of a clock x like so:
![image](https://github.com/UPPAALModelChecker/UPPAAL-Meta/assets/10059450/52bb9586-7299-4d57-8e4f-aed38b62f869)

**Reply by mikucionisaau** [reply]
Perhaps we could add an option to specify the number of edge-transitions for Zeno detection?

Right now the bound hardcoded, but we could use a floating point number to encode it, and then the user may specify `inf` for infinity and effectively disable the Zeno detector.
